python/triton/common/backend.py

```diff
@@ -174,6 +174,10 @@ def get_cuda_version_key():
     global _cached_cuda_version_key
     if _cached_cuda_version_key is None:
         key = compute_core_version_key()
-        ptxas = path_to_ptxas()[0]
-        _cached_cuda_version_key = key + '-' + hashlib.sha1(subprocess.check_output([ptxas, "--version"])).hexdigest()
+        try:
+            ptxas = path_to_ptxas()[0]
+            ptxas_version = subprocess.check_output([ptxas, "--version"])
+        except RuntimeError:
+            ptxas_version = b"NO_PTXAS"
+        _cached_cuda_version_key = key + '-' + hashlib.sha1(ptxas_version).hexdigest()
```

Comment:
I had an impression that `get_cuda_version_key()` was called even for cpu-only users, so I just modified the function rather than eliminating references to cuda in the cpu-only branches.

This `try-except` block has the adverse side-effect that users who actually want to use `ptxas` won't see this `RuntimeError`, but instead will observe some failure later. I think they'll see the same error upon a call to `llir_to_ptx` or `ptx_to_cubin`, so it's probably fine.

Reply:
I see now that in `v2.0.0` you had been doing effectively the same: catching the `Exception` and resetting `ptxas_version` to a fixed (empty) string literal. Was there a reason not to do that anymore?